akka-projection-kafka/src/main/scala/akka/projection/kafka/KafkaSourceProvider.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.kafka
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.kafka.ConsumerSettings
+import akka.kafka.Subscriptions
+import akka.kafka.scaladsl.Consumer
+import akka.kafka.scaladsl.MetadataClient
+import akka.projection.internal.MergeableOffsets
+import akka.projection.internal.MergeableOffsets.SurrogateProjectionKey
+import akka.projection.scaladsl.SourceProvider
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+
+object KafkaSourceProvider {
+  private type ReadOffsetsHandler = () => Future[Option[MergeableOffsets.Offset[Long]]]
+  private val RegexTp = """(.+)-(\d+)""".r
+  private val KafkaMetadataTimeout = 10.seconds // TODO: get from config
+
+  /**
+   * Create a [[SourceProvider]] that resumes from externally managed offsets
+   */
+  def apply[K, V](
+      systemProvider: ClassicActorSystemProvider,
+      settings: ConsumerSettings[K, V],
+      topics: Set[String]): SourceProvider[MergeableOffsets.Offset[Long], ConsumerRecord[K, V]] =
+    new KafkaSourceProvider[K, V](systemProvider, settings, topics)
+}
+
+@InternalApi
+class KafkaSourceProvider[K, V] private[kafka] (
```

Comment:
move this to `internal` package, and add the usual 
```
/**
 * INTERNAL API
 */
@InternalApi private[akka] 
```

Reply:
:+1: